contracts/teller-network/Fees.sol

```diff
@@ -36,45 +34,66 @@ contract Fees is Ownable {
      * @dev Can only be called by the owner of the contract
      *      TODO: if the contract will be changed to remove ownership, remove this function
      */
-    function setFeeDestinationAddress(address _addr) public onlyOwner {
+    function setFeeDestinationAddress(address payable _addr) public onlyOwner {
         feeDestination = _addr;
         emit FeeDestinationChanged(_addr);
     }
 
     /**
      * @notice Set Fee Amount
-     * @param _amount New Amount
+     * @param _feeMilliPercent New millipercent
      * @dev Can only be called by the owner of the contract
      *      TODO: if the contract will be changed to remove ownership, remove this function
      */
-    function setFeeAmount(uint _amount) public onlyOwner {
-        feeAmount = _amount;
-        emit FeeAmountChanged(_amount);
+    function setFeeAmount(uint _feeMilliPercent) public onlyOwner {
+        feeMilliPercent = _feeMilliPercent;
+        emit FeeMilliPercentChanged(_feeMilliPercent);
     }
 
     /**
      * @notice Withdraw fees by sending them to the fee destination address
      */
-    function withdrawFees() public {
-        uint fees = feeBalance;
-        feeBalance = 0;
-        require(feeToken.transfer(feeDestination, fees), "Error transfering fees");
-        emit FeesWithdrawn(fees);
+    function withdrawFees(address _tokenAddress) public {
+        uint fees = feeTokenBalances[_tokenAddress];
+        feeTokenBalances[_tokenAddress] = 0;
+        if (_tokenAddress == address(0)) {
+            require(address(this).balance >= fees, "Not enough balance");
+            feeDestination.transfer(fees);
+            emit MyEvent(fees, feeDestination, address(this).balance);
+        } else {
+            ERC20Token tokenToWithdraw = ERC20Token(_tokenAddress);
+            require(tokenToWithdraw.transfer(feeDestination, fees), "Error transferring fees");
+        }
+        emit FeesWithdrawn(fees, _tokenAddress);
+    }
+
+    function getFeeFromAmount(uint _value) internal returns(uint) {
+        // To get the factor, we divide like 100 like a normal percent, but we multiply that by 1000 because it's a milliPercent
+        // Eg: 1 % = 1000 millipercent => Factor is 0.01, so 1000 divided by 100 * 1000
+        return (_value * feeMilliPercent) / (100 * 1000);
     }
 
     /**
      * @notice Pay fees for a transaction or element id
      * @param _from Address from where the fees are being extracted
      * @param _id Escrow id or element identifier to mark as paid
+     * @param _value Value sold in the escrow
+     * @param _feeAmount Fee amount calculated by the front-end
+     * @param _tokenAddress Address of the token sold in the escrow
      * @dev This will only transfer funds if the fee  has not been paid
      */
-    function payFee(address _from, uint _id) internal {
-        if(feePaid[_id]) return;
+    function payFee(address _from, uint _id, uint _value, uint _feeAmount, address _tokenAddress) internal {
```

Comment:
I'm confused on why the _feeAmount must be an input value, instead of the contract just receiving an amount and doing the fee split on its own.

Reply:
Yea....... I think you're right and I don't need it at all haha
At first, I thought I needed to pass it so that the Fees contract could make sure the front-end sent the right amount, but it's all in the allowance or the value passed, so it will already be reverted if the user didn't allow or pass enough value.
Good catch, will change that